fab3/ethservice.go

```diff
@@ -531,21 +535,36 @@ func (s *ethService) NewFilter(_ *http.Request, filter *types.GetLogsArgs, resul
 	s.filterMapLock.Lock()
 	s.filterSeq++
 	index := s.filterSeq
-	s.filterMap[index] = filter
+	s.filterMap[index] = &logsFilter{lastAccessTime: time.Now(), logArgs: filter}
 	s.filterMapLock.Unlock()
 	*result = "0x" + strconv.FormatUint(index, 16)
 	return nil
 }
 
-func (s *ethService) UninstallFilter(_ *http.Request, filterID *string, removed *bool) error {
-	id, err := strconv.ParseUint(strip0x(*filterID), 16, 64)
+func (s *ethService) NewBlockFilter(_ *http.Request, _ *interface{}, result *string) error {
+	latestExistingBlock, err := s.parseBlockNum("latest")
 	if err != nil {
-		return errors.Wrap(err, "failed to parse filter id")
+		return errors.Wrap(err, "latest block number not available at this time")
 	}
 
+	s.filterMapLock.Lock()
+	s.filterSeq++
+	index := s.filterSeq
+
+	s.filterMap[index] = &newBlockFilter{latestBlockSeen: latestExistingBlock, lastAccessTime: time.Now()}
+	s.filterMapLock.Unlock()
+	*result = "0x" + strconv.FormatUint(index, 16)
+	return nil
+}
+
+func (s *ethService) UninstallFilter(_ *http.Request, filterID *types.FilterID, removed *bool) error {
+	id := filterID.ID
 	s.filterMapLock.Lock()
 	defer s.filterMapLock.Unlock()
 
+	// do we care about timing out filters while removing them?
+	// i.e. turning the successful delete if present into an error
+	// because it timed out?
 	if _, ok := s.filterMap[id]; ok {
 		delete(s.filterMap, id)
 		*removed = true
```

Comment:
that's basically what the comment above says.

I don't think there's any diagnostic information to be gained by erroring on timeout vs returning false that the filter was not present.

I also don't think there's much information in the single bit of true/false we're giving back for "we found it and removed it" vs "we didn't find it" vs "we found it, and it was timed out".

I can't very well conceive of anything harmful in the current behavior.